### ISSUE TITLE
Error text when using az login and wrong tenant

### ISSUE THREAD

**Issue by dozer75** [Azure/azure-cli#9115]
I have struggled using a service-principal login today, but the error on wrong tenant is wrong and caused confusion and frustration. 

It states:

> The command failed with an unexpected error. Here is the traceback:
> 
> The authority url must be of the format **https://login.microsoftonline.com/your_tenant**
> Traceback (most recent call last):
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\knack\knack\cli.py", line 206, in invoke
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 326, in execute
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 384, in _run_jobs_serially
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 377, in _run_job
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\six\six.py", line 693, in reraise
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 354, in _run_job
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 145, in __call__
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\__init__.py", line 451, in default_command_handler
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-profile\azure\cli\command_modules\profile\custom.py", line 133, in login
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 204, in find_subscriptions_on_login
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 805, in find_from_service_principal_id
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 821, in _create_auth_context
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 97, in _authentication_context_factory
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\adal\adal\authentication_context.py", line 88, in __init__
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\adal\adal\authority.py", line 49, in __init__
>   File "C:\Users\<VSADMUSER>\AppData\Local\Temp\pip-install-z6u9vbmn\adal\adal\authority.py", line 81, in _validate_authority_url
> ValueError: The authority url must be of the format https://login.microsoftonline.com/your_tenant

But, the tenant shall NOT be in the format `https://login.microsoftonline.com/your_tenant`  it's just the tenant id without `https://login.microsoftonline.com/`. It would also be nice that it says `tenant_id` instead of `your_tenant`. 

Using azure-cli 2.0.62

**Comment by yugangw-msft** [comment]
The `your_tenant` can either be a domain name like `mycompany.onmicrosoft.com` or a GUID style of tenant id; either the error should be more clear. Can you please open an bug at https://github.com/AzureAD/azure-activedirectory-library-for-python?

**Comment by rayluo** [comment]
@dozer75 
Wait a minute. Could it be possible that, you somehow already got an impression from SOMEWHERE ELSE, that you would need to use a full url "https://login.microsoftonline.com/your_tenant_name" as your tenant name? If that is the case, let's know where that "SOMEWHERE ELSE" is, and try to clean it up there.

From Azure CLI's underlying library [ADAL Python's perspective](https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/1.2.1/adal/authentication_context.py), historically we don't even have an explicit tenant concept; we just accepts a full authority string in the form of `https://login.microsoftonline.com/your_tenant`. If for whatever reason the input becomes something like `https://login.microsoftonline.com/https://login.microsoftonline.com/your_tenant`, ADAL Python would raise the exception you saw, and rightfully so.

@yugangw-msft Perhaps Azure CLI can have a safe guard on the `--tenant XYZ` parameter to detect some input which does not look like a tenant name?

**Comment by dozer75** [comment]
@raylou No, it was the error message that confused me. Since I didn't find any good description of the parameter then i tried with a bogus text first to see what the error was, then I got that message and tried all kinds of combinations starting with https://login.microsoft.online. After @yugangw-msft answered me I found a documentation that stated what he said.

I would agree that Azure CLI could have a safeguard too, however the python script could also have the correct message from the beginning.

It would also be nice if the (documentation)[https://docs.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest#az-login] had a description of what the tenant should be.
